src/mame/includes/tsconf.h

```diff
@@ -0,0 +1,212 @@
+// license:BSD-3-Clause
+// copyright-holders:Andrei I. Holub
+/*****************************************************************************
+ *
+ * includes/tsconfig.h
+ *
+ ****************************************************************************/
+#ifndef MAME_INCLUDES_TSCONF_H
+#define MAME_INCLUDES_TSCONF_H
+
+#pragma once
+
+#include "emu.h"
```

Comment:
You must not put this in headers – it needs to be included as the first thing in each translation unit and nowhere else, or you’ll get issues with different build configurations (with/without precompiled headers, with/without implicit prefix header).